Tests/MongoSwiftTests/Document+CollectionTests.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+@testable import MongoSwift
+import Nimble
+import XCTest
+
+final class Document_CollectionTests: MongoSwiftTestCase {
```

Comment:
This is a little tricky. Swift `Collection` (including `Dictionary`) will throw fatal error for doing things like `removeFirst()` on an empty collection (and our `Document` does this as well already). This means its pretty much impossible to test for this since fatal errors cannot be caught in tests without considerable hacky work.

I've at least added tests for the things I can check (`startIndex` and `endIndex`), but the other methods like `subscript` and `index(after:)` can't really be tested.

I did at least try them by hand to see if we are consistent, and they both throw a fatal error just like we do. In fact, we do a better job because we actually include a message for why its failing 😄!

Also nit: removed variable name stutter by renaming `testDocumentIndexLogic` -> `testIndexLogic`,

Reply:
perfect, that was all I wanted to know.